deploy/sdk/src/dynamo/sdk/cli/build.py

```diff
@@ -128,6 +128,11 @@ def from_service(cls, service: ServiceInterface[T]) -> ServiceInfo:
         for ep_name, endpoint in service.get_dynamo_endpoints().items():
             if DynamoTransport.HTTP in endpoint.transports:
                 api_endpoints.append(f"/{ep_name}")
+        http_exposed = service.config.http_exposed
+        if len(api_endpoints) > 0:
+            assert (
+                http_exposed
+            ), "HTTP exposed must be True if API endpoints are defined"
```

Comment:
I don't think we need to/should require this actually. I think it's reasonable for the user to want to add API endpoints but not expose HTTP, either because they don't have ingress/istio or they just don't want to and would prefer portforwarding